app/controllers/vendors_controller.rb

```diff
@@ -0,0 +1,54 @@
+class VendorsController < ApplicationController
+
+  def index
+    @vendors = Vendor.order(vendor_name: :asc)
+    render :index
+  end
+
+  def show
+    @vendors = Vendor.where(id: params[:id])
+    render :single_vendor_view
+  end
+
+  def new
+    @new_vendor = Vendor.new
+    @market = Market.where(id: params[:id]).first
```

Comment:
Market.where will return an array, which is why you have to chain the .first method on it.

A simpler approach is to do Market.find(params[:id]) because the find method only returns one object.